transport.go

```diff
@@ -777,6 +777,9 @@ var InitialTransport = http.DefaultTransport
 // than http.DefaultClient).
 var oldClients = map[*http.Client]http.RoundTripper{}
 
+// oldClientsLock protects oldClients from concurrent writes.
+var oldClientsLock = &sync.Mutex{}
```

Comment:
Sorry last thing, no need of a pointer here:
```suggestion
var oldClientsLock sync.Mutex
```

Reply:
No worries :) Fixed.